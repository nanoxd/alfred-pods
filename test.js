import test from 'ava'
import alfyTest from 'alfy-test'
import { podStanza } from './index'

test('Empty Results', async t => {
  const alfy = alfyTest()
  const query = 'REDFASFDLAKJ;FAS'
  const result = await alfy(query)

  t.deepEqual(result, [
    {
      title: 'No Pods Found',
      subtitle: 'Open CocoaPods Search',
      arg: `https://cocoapods.org/?q=${query}`
    }
  ])
})

test('Valid Results', async t => {
  const alfy = alfyTest()
  const query = 'Alamofire'
  const result = await alfy(query)

  const alamofire = result[0]

  t.snapshot(alamofire)
})

test('Pod Stanza', () => {
  const pod = { id: 'Alamofire' }
  t.deepEqual(podStanza(pod), `pod '${pod.id}'`)
})
