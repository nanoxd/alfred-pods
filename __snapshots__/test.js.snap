// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[anonymous] 1`] = `
Object {
  "arg": "https://github.com/Alamofire/Alamofire",
  "icon": Object {
    "path": " ",
  },
  "subtitle": "Elegant HTTP Networking in Swift",
  "text": Object {
    "copy": "pod 'Alamofire'",
  },
  "title": "Alamofire (4.4.0)",
}
`;

exports[`Valid Results 1`] = `
Object {
  "arg": "https://github.com/Alamofire/Alamofire",
  "subtitle": "Elegant HTTP Networking in Swift",
  "text": Object {
    "copy": "pod 'Alamofire'",
  },
  "title": "Alamofire (4.4.0)",
}
`;
